### PR TITLE
[Plugin] Plural Kit Integration

### PR DESCRIPTION
**Pull request by Scyye**
Fork of [PluralChum](https://github.com/estroBiologist/pluralchum) for Vencord

Current features include:
1. Edit button
2. Member IDs in chat
3. Member colour in chat

Planned features:
1. `PK` badge, compatible with moreUserTags
2. Functional member-specific profile popout
3. Blocking 
4. Reproxy modal
5. Option of role colours rather than member colour

There's a few things i dont fully understand how to use (ex. `DataStore`), if I used that, or anything else incorrectly, or theres any changes needed, please let me know and ill be sure to fix them.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
         "gifenc": "github:mattdesl/gifenc#64842fca317b112a8590f8fef2bf3825da8f6fe3",
         "monaco-editor": "^0.43.0",
         "nanoid": "^4.0.2",
-        "virtual-merge": "^1.0.1"
+        "virtual-merge": "^1.0.1",
+        "pkapi.js": "7.2.1"
```
Comment (MrDiamondDog) [inline package.json:49]: Plugins should not add dependencies. This package is just a wrapper around an api, so use that instead

File: src/plugins/pluralKit/index.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        default: false
+    }
+});
+
+// I dont fully understand how to use datastores, if i used anything incorrectly please let me know
+const DATASTORE_KEY = "pk";
+
+let authors: Record<string, { messageIds: string[]; member: Member|string|undefined; system: System; }> = {};
+(async () => {
+    authors = await DataStore.get<Record<string, { name: string; messageIds: string[]; member: Member|string|undefined; system: System; }>>(DATASTORE_KEY) || {};
```
Comment (MrDiamondDog) [inline src/plugins/pluralKit/index.tsx:61]: Put this in your plugin's start function

File: src/plugins/pluralKit/index.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        type: OptionType.BOOLEAN,
+        description: "Display member IDs in chat",
+        default: false
+    }
+});
+
+// I dont fully understand how to use datastores, if i used anything incorrectly please let me know
+const DATASTORE_KEY = "pk";
+
+let authors: Record<string, { messageIds: string[]; member: Member|string|undefined; system: System; }> = {};
```
Comment (MrDiamondDog) [inline src/plugins/pluralKit/index.tsx:59]: Please make this a type instead of typing it out everytime

File: src/plugins/pluralKit/index.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+(async () => {
+    authors = await DataStore.get<Record<string, { name: string; messageIds: string[]; member: Member|string|undefined; system: System; }>>(DATASTORE_KEY) || {};
+})();
+
+
+export default definePlugin({
+    name: "Plural Kit",
```
Comment (MrDiamondDog) [inline src/plugins/pluralKit/index.tsx:66]: Your plugin's name spills follow our naming convention, so it should be `PluralKit`

File: src/plugins/pluralKit/index.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            return <span style={{
+                color: `#${color}`,
+            }}>{prefix}{member.display_name??member.name}{authorOfMessage.system?.tag??""}{settings.store.displayMemberId?` (${member.id})`:""}</span>;
+        } catch {
+            return <>{prefix}{author?.nick}</>;
+        }
+    },
+
+    async start() {
+        DataStore.createStore(DATASTORE_KEY, DATASTORE_KEY);
```
Comment (MrDiamondDog) [inline src/plugins/pluralKit/index.tsx:104]: You don't need to do this

File: src/plugins/pluralKit/index.tsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+        addButton("pk-edit", msg => {
+            if (!msg) return null;
+            console.log(msg);
```
Comment (MrDiamondDog) [inline src/plugins/pluralKit/index.tsx:108]: make sure you get rid of these

**Review by MrDiamondDog — CHANGES_REQUESTED** [review]
This plugin seems kinda niche but I'll leave this review

**Comment by Vendicated** [comment]
thanks for your contribution!

unfortunately this is way too niche, sorry

**Comment by Scyye** [comment]
> thanks for your contribution!
> 
> unfortunately this is way too niche, sorry
> @Vendicated 

if i could appeal my ban from the discord, would there be anywhere for me to share it there?

**Comment by Scyye** [comment]
@Vendicated ?

**Comment by lewisakura** [comment]
You aren't banned from the Discord server, according to the account you're using as your developer account. Either way, this is not the appropriate place to file such an appeal.

**Comment by Scyye** [comment]
> You aren't banned from the Discord server, according to the account you're using as your developer account. Either way, this is not the appropriate place to file such an appeal.

i wasnt attempting to appeal here, my hope was i would receive a friend request or something of the sort, and i could do it through DMs; where would be the proper place?

**Comment by c45513v** [comment]
> thanks for your contribution!
> 
> unfortunately this is way too niche, sorry

For the record, a lot of us who would use this plugin see your reply as pure bigotry. I know in a previous response it was mentioned that its not common in the servers that you as developers are in- That may be true, but that's an extremely selfcentered point of view. I highly implore you to reconsider.

**Comment by lewisakura** [comment]
@c45513v Please read #3058 with an updated comment, and please note that we do not accept plugins that only work with specific bots under the latest [plugin rules](https://github.com/Vendicated/Vencord/blob/main/CONTRIBUTING.md#plugin-rules).

**Pull request closed without merging**